src/stream/src/executor/top_n/top_n_cache.rs

```diff
@@ -309,38 +310,46 @@ impl TopNCacheTrait for TopNCache<false> {
 impl TopNCacheTrait for TopNCache<true> {
     fn insert(
         &mut self,
-        ordered_pk_row: OrderedRow,
+        cache_key: CacheKey,
         row: Row,
         res_ops: &mut Vec<Op>,
         res_rows: &mut Vec<CompactedRow>,
     ) {
         assert!(self.low.is_empty());
 
-        let elem_to_compare_with_middle = (ordered_pk_row, row);
+        let elem_to_compare_with_middle = (cache_key, row);
 
         if !self.is_middle_cache_full() {
             self.middle.insert(
-                elem_to_compare_with_middle.0,
+                elem_to_compare_with_middle.0.clone(),
                 (&elem_to_compare_with_middle.1).into(),
             );
             res_ops.push(Op::Insert);
             res_rows.push((&elem_to_compare_with_middle.1).into());
             return;
         }
 
-        let sort_key = elem_to_compare_with_middle.0.prefix(self.order_by_len);
+        let sort_key = &elem_to_compare_with_middle.0 .0;
```

Comment:
.0.0 doesn't look very nice... Use named fields instead?

Reply:
I prefer keep `CacheKey` a type instead of a struct,  although `0 .0` is weird.